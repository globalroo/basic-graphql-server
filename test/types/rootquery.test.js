import {
	GraphQLList,
} from "graphql";

jest.mock("../../src/data/data-api", () => ({
	getVideoById: jest.fn(),
	getVideoByTitle: jest.fn(),
	getVideosByAspect: jest.fn(),
	getVideos: jest.fn()
}));

import * as rootQuery from "../../src/types/root-query";
import { VideoType } from "../../src/types/video-type";

import {
	getVideoById,
	getVideoByTitle,
	getVideosByAspect,
	getVideos
} from "../../src/data/data-api";

describe("GraphQL Resolvers: getVideoQuery", () => {
	afterEach(() => {
		getVideoById.mockClear();
		getVideoByTitle.mockClear();
	});
	test("getVideoQuery returns correct type", () => {
		const queryObj = rootQuery.getVideoQuery();
		expect (queryObj.type).toEqual(VideoType);
	});
	test("getVideoQuery expects correct args", () => {
		const queryObj = rootQuery.getVideoQuery();
		expect (queryObj.args).toMatchSnapshot();
	});
	test("getVideoQuery resolver calls correct functions: getVideoById", () => {
		const queryObj = rootQuery.getVideoQuery();
		queryObj.resolve(undefined, { id: 1 });
		expect(getVideoById).toHaveBeenCalled();
	});
	test("getVideoQuery resolver calls correct functions: getVideoByTitle", () => {
		const queryObj = rootQuery.getVideoQuery();
		queryObj.resolve(undefined, { title: "Minions" });
		expect(getVideoByTitle).toHaveBeenCalled();
	});
	test("getVideoQuery resolver calls correct functions: getVideoByTitle", () => {
		const queryObj = rootQuery.getVideoQuery();
		queryObj.resolve(undefined, { });
		expect(getVideoByTitle).not.toHaveBeenCalled();
	});
});

describe("GraphQL Resolvers: getAspectQuery", () => {
	afterEach(() => {
		getVideosByAspect.mockClear();
	});
	test("getAspectQuery returns correct type", () => {
		const queryObj = rootQuery.getAspectQuery();
		expect (queryObj.type).toEqual(new GraphQLList(VideoType));
	});
	test("getAspectQuery expects correct args", () => {
		const queryObj = rootQuery.getAspectQuery();
		expect (queryObj.args).toMatchSnapshot();
	});
	test("getAspectQuery resolver calls correct functions: getVideosByAspect", () => {
		const queryObj = rootQuery.getAspectQuery();
		queryObj.resolve(undefined, { aspect: "16x9" });
		expect(getVideosByAspect).toHaveBeenCalled();
	});
});

describe("GraphQL Resolvers: getVideosQuery", () => {
	afterEach(() => {
		getVideos.mockClear();
	});
	test("getVideosQuery returns correct type", () => {
		const queryObj = rootQuery.getVideosQuery();
		expect (queryObj.type).toEqual(new GraphQLList(VideoType));
	});
	test("getVideosQuery expects correct args", () => {
		const queryObj = rootQuery.getVideosQuery();
		expect (queryObj.args).toMatchSnapshot();
	});
	test("getVideosQuery resolver calls correct functions: getVideosByAspect", () => {
		const queryObj = rootQuery.getVideosQuery();
		queryObj.resolve(undefined, {});
		expect(getVideos).toHaveBeenCalled();
	});
});
