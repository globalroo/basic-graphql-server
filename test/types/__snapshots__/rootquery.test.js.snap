// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL Resolvers: getAspectQuery getAspectQuery expects correct args 1`] = `
Object {
  "aspect": Object {
    "description": "The Aspect required",
    "type": "String!",
  },
}
`;

exports[`GraphQL Resolvers: getVideoQuery getVideoQuery expects correct args 1`] = `
Object {
  "id": Object {
    "description": "The ID",
    "type": "Int",
  },
  "title": Object {
    "description": "The Title",
    "type": "String",
  },
}
`;

exports[`GraphQL Resolvers: getVideosQuery getVideosQuery expects correct args 1`] = `
Object {
  "aspect": Object {
    "description": "Optional Aspect",
    "type": "String",
  },
}
`;
